lib/Provider/MailProvider.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\Mail\Provider;
+
+use OCA\Mail\Account;
+use OCA\Mail\Service\AccountService;
+use OCP\Mail\Provider\Address as MailAddress;
+use OCP\Mail\Provider\IProvider;
+use OCP\Mail\Provider\IService;
+use Psr\Container\ContainerInterface;
+use Psr\Log\LoggerInterface;
+
+class MailProvider implements IProvider {
+
+	public function __construct(
+		protected ContainerInterface $container,
+		protected AccountService $accountService,
+		protected LoggerInterface $logger
+	) {
+	}
+
+	/**
+	 * Arbitrary unique text string identifying this provider
+	 *
+	 * @since 4.0.0
+	 *
+	 * @return string id of this provider (e.g. UUID or 'IMAP/SMTP' or anything else)
+	 */
+	public function id(): string {
+		return 'mail-application';
+	}
+
+	/**
+	 * Localized human friendly name of this provider
+	 *
+	 * @since 4.0.0
+	 *
+	 * @return string label/name of this provider (e.g. Plain Old IMAP/SMTP)
+	 */
+	public function label(): string {
+		return 'Mail Application';
+	}
```

Comment:
not sure if this is even visible to the user but there is no localization

Reply:
Done